lib/rspec/expectations/handler.rb

```diff
@@ -69,7 +69,7 @@ class NegativeExpectationHandler
       def self.handle_matcher(actual, initial_matcher, message=nil, &block)
         ExpectationHelper.with_matcher(self, initial_matcher, message) do |matcher|
           return ::RSpec::Matchers::BuiltIn::NegativeOperatorMatcher.new(actual) unless initial_matcher
-          !(does_not_match?(matcher, actual, &block) || ExpectationHelper.handle_failure(matcher, message, :failure_message_when_negated))
+          does_not_match?(matcher, actual, &block) || ExpectationHelper.handle_failure(matcher, message, :failure_message_when_negated)
```

Comment:
Can someone check my sanity here? If the spec descriptions hold true, what was here doesn't return the matcher value (and it should). If not, then the spec descriptions are wrong and `match`, `matches?`, `does_not_match?`, etc should be returning booleans...I think.